plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/PropertiesSecretSource.java

```diff
@@ -0,0 +1,56 @@
+package io.jenkins.plugins.casc.impl.secrets;
+
+import hudson.Extension;
+import io.jenkins.plugins.casc.SecretSource;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This {@link SecretSource} implementation allows to use a .properties file for providing secrets.
+ * The default file path is {@code /run/secrets/secrets.properties}, which can be changed via
+ * {@code SECRETS} environment variable.
+ *
+ * @author <a href="mailto:d.estermann.de@gmail.com">Daniel Estermann</a>
+ */
+@Extension
+public class PropertiesSecretSource extends SecretSource {
+
+    private static final Logger LOGGER = Logger.getLogger(PropertiesSecretSource.class.getName());
+
+    /**
+     * Default path for .properties file
+     */
+    public static final String SECRETS_DEFAULT_PATH = "/run/secrets/secrets.properties";
+
+    private final Properties secrets = new Properties();
+
+    public PropertiesSecretSource() {
```

Comment:
Would be nice to run the retrieval upon first request instead of doing it through constructor. It would allow this code to be invoked on-demand, especially if the extension is disabled later